srcCs/RTIDDSImpl.cs

```diff
@@ -2070,17 +2043,17 @@ public IMessagingReader CreateReader(string topic_name, IMessagingCB callback)
                 }
             }
 
-            if ("ThroughputQos" == qos_profile ||
-                "NoAckThroughputQos" == qos_profile) {
-                dr_qos.durability.kind = (DDS.DurabilityQosPolicyKind)_Durability;
-                dr_qos.durability.direct_communication = _DirectCommunication;
+            if (!_UsePositiveAcks 
+                    && (qos_profile == "ThroughputQos"
+                            || qos_profile == "LatencyQos"))
+            {
+                dr_qos.protocol.disable_positive_acks = true;
             }
 
-            if (("LatencyQos" == qos_profile ||
-                "NoAckLatencyQos" == qos_profile) &&
-                !_DirectCommunication &&
-                (_Durability == 2 ||
```

Comment:
why dont we use:
- DDS::DurabilityQosPolicyKind::TRANSIENT_DURABILITY_QOS instead of 2
- DDS::DurabilityQosPolicyKind::PERSISTENT_DURABILITY_QOS  instead of 3

REF: https://community.rti.com/static/documentation/connext-dds/5.3.0/doc/api/connext_dds/api_dotnet/group__DDSDurabilityQosModule.html

Reply:
Done:

```
            if ("LatencyQos" == qos_profile
                    && !_DirectCommunication
                    && ((DDS.DurabilityQosPolicyKind)_Durability 
                            == DDS.DurabilityQosPolicyKind.TRANSIENT_DURABILITY_QOS
                        || (DDS.DurabilityQosPolicyKind)_Durability 
                            == DDS.DurabilityQosPolicyKind.PERSISTENT_DURABILITY_QOS))
            {
```